javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -0,0 +1,1761 @@
+
+  @inline
+  private def charEqualTo(c: Char, cs: Char*): Boolean = cs.contains(c)
+
+  @inline
+  private def insertString(s: String, pos: Int, s2: String): String =
+    s.substring(0, pos) + s2 + s.substring(pos)
+
+  @inline
+  private def insertString(s: String, pos: Int, s2: String, s2Start: Int,
+      s2Len: Int): String =
+    insertString(s, pos, s2.substring(s2Start, s2Start + s2Len))
+
+  private implicit class StringOps(val s: String) extends AnyVal {
+    @inline
+    def insert(pos: Int, s2: String): String = insertString(s, pos, s2)
+
+    @inline
+    def insert(pos: Int, s2: String, s2Start: Int, s2Len: Int): String =
+      insertString(s, pos, s2, s2Start, s2Len)
+  }
+
+  @inline
+  private final class QuotAndRem[T](val quot: T, val rem: T)
+
+  import scala.language.implicitConversions
+  import scala.reflect.ClassTag
+
+  @inline
+  private implicit def toArray[T: ClassTag](qr: QuotAndRem[T]): Array[T] =
+    Array[T](qr.quot, qr.rem)
+
+  private implicit def toQuotAndRem[T](dr: Array[T]): QuotAndRem[T] =
+    new QuotAndRem[T](dr(0), dr(1))
+}
+
+class BigDecimal() extends Number with Comparable[BigDecimal] {
+  import BigDecimal._
+  import Multiplication._
+
+  /** The <code>String</code> representation is cached. */
+  private var _toStringImage: String = null
+
+  /** Cache for the hash code. */
+  private var _hashCode: Int = 0
+
+  /** The internal representation of {@code BigDecimal}.
+   *
+   *  The arbitrary precision integer (unscaled value) in the internal
+   *  representation of {@code BigDecimal}.
+   */
+  private var _intVal: BigInteger = _
+
+  private var _bitLength: Int = 0
+
+  private var _smallValue: Long = 0
+
+  /** The 32-bit integer scale in the internal representation of {@code BigDecimal}. */
+  private var _scale: Int = 0
+
+  /** Represent the number of decimal digits in the unscaled value.
+   *
+   *  This precision is calculated the first time, and used in the following calls
+   *  of method <code>precision()</code>. Note that some call to the private
+   *  method <code>inplaceRound()</code> could update this field.
+   *
+   *  @see #precision()
+   *  @see #inplaceRound(MathContext)
+   */
+  private var _precision: Int = 0
+
+  private def this(smallValue: Long, scale: Int) {
+    this()
+    _smallValue = smallValue
+    _scale = scale
+    _bitLength = bitLength(smallValue)
+  }
+
+  private def this(smallValue: Int, scale: Int) {
+    this()
+    _smallValue = smallValue
+    _scale = scale
+    _bitLength = bitLength(smallValue)
+  }
+
+  def this(in: Array[Char], offset: Int, len: Int) {
+    this()
+
+    val last = offset + len - 1 // last index to be copied
+
+    if (in == null)
+      throw new NullPointerException("in == null")
+
+    if (last >= in.length || offset < 0 || len <= 0 || last < 0)
+      throw new NumberFormatException(
+        s"Bad offset/length: offset=${offset} len=$len in.length=${in.length}")
```

Comment:
This line should be indented 4 spaces with respect to the `throw` line.

Also, since the body of the `if` is not a one-liner, it should be enclosed with `{}`